tsdb/engine/tsm1/wal.go

```diff
@@ -27,6 +27,9 @@ const (
 
 	defaultBufLen = 1024 << 10 // 1MB (sized for batches of 5000 points)
 
+	// walEncodeBufSize is the size of the wal entry encoding buffer
+	walEncodeBufSize = 4 * 1024 * 1024
```

Comment:
It's the default size of the buffers stored in a `sync.Pool` instance used during wal entry marshaling.  The `sync.Pool` is shared across all shards.  It's basically a free list of temporary buffers that get re-used. 

See: https://golang.org/pkg/sync/#Pool